### PR TITLE
Adds the play button to the slider

### PR DESCRIPTION
**Pull request by anmoljagetia**
This is in reference to #353.
@GordonSmith @mlzummo  Please review.
This adds a play and a loop button to the slider.

The buttons don't appear as of now on the rawgit link, becuase the styles override the FontAwesome style with Verdana. #406 fixes this issue.

Demo at : http://rawgit.com/anmoljagetia/Visualization/playSlider2/demos/dermatology.html

Also, I am looking for a better commit message, and squashing all the commits into one? does "Adds the ability to play or loop GH-353" look plausible?

### CODE REVIEW

File: demos/dermatology.html [hunk 1]
```diff
@@ -42,6 +42,7 @@
                     <li>
                         <prompt>Widget:</prompt>
                         <select id="widgetSelect" onchange="testWidget(this.value);">
+                            <option value="src/other/Slider">Slider</option>
```
Comment (GordonSmith) [inline demos/dermatology.html]: Move this down to the other "Other" widgets,

File: src/other/Slider.css [hunk 2]
```diff
@@ -35,3 +35,28 @@
     stroke-width: 1.00px;
     pointer-events: none;
 }
+
+.common_Icon:hover {
```
Comment (GordonSmith) [inline src/other/Slider.css]: Should these styles be in the common/Icon.css file?

File: src/other/Slider.js [hunk 3]
```diff
@@ -62,6 +107,47 @@
...
+                    ;
+                    context._playing = false;
+                    
+                    clearInterval(context.intervalHandler);
+                }
+            }
+        }, context.playInterval());
+    };
+
+    Slider.prototype.pause = function () {
```
Comment (GordonSmith) [inline src/other/Slider.js]: Remove unused code (it has a tendency to never get updated...)

File: src/other/Slider.js [hunk 4]
```diff
@@ -62,6 +107,47 @@
...
+        // Will be added later
+    };
+
+    Slider.prototype.stop = function () {
+        console.log("stopped");
```
Comment (GordonSmith) [inline src/other/Slider.js]: Remove all console messages (except the default click)

File: src/other/Slider.js [hunk 5]
```diff
@@ -109,20 +198,67 @@
...
         var context = this;
-        var width = this.width() - 50;  //TODO - 50 should be "padding"
+        var width = this.width() - 100;  //TODO - 50 should be "padding"
+        // var height = this.height() - 20;  //TODO - 20 should be "padding"
 
         this.xScale
             .domain([this.low(), this.high()])
             .range([-width/2, width/2])
         ;
+         this.data(this._data);
```
Comment (GordonSmith) [inline src/other/Slider.js]: Your setting the current data with the current data?

File: src/other/Slider.js [hunk 6]
```diff
@@ -109,20 +198,67 @@
...
+            this._icon1
+                .display(true)
+                .render()
+            ;
+
+            this.xScale
+                .domain([this.low(), this.high()])
+                .range([-width/2, width/2 - this._icon.diameter()*2 - this.playGutter()])
+            ;
+            this.data(this._data);
```
Comment (GordonSmith) [inline src/other/Slider.js]: Your setting the current data with the current data?

File: src/other/Slider.js [hunk 7]
```diff
@@ -109,20 +198,67 @@
             .attr("class", "handle")
             .attr("transform", "translate(0,-27)")
         ;
+
+        this._icon
```
Comment (GordonSmith) [inline src/other/Slider.js]: These should be in the enter method.

File: src/other/Slider.css (hunk 2 above)
Comment (anmoljagetia) [inline src/other/Slider.css]: Actually they are only meant to work on the icon here, I don't want to override the default icon settings, in the main icon styles file.

File: src/other/Slider.js (hunk 3 above)
Comment (anmoljagetia) [inline src/other/Slider.js]: are you referring to the pause method?

**Comment by GordonSmith** [comment]
Looks fine.

**Comment by mlzummo** [comment]
LGTM

**Comment by GordonSmith** [comment]
In your commit you had "Fixes GH #353" - I am not sure if that will auto close the issue, I know:
Fixes GH-353
and
Fixes #353 
Will.

**Pull request merged**